### ISSUE TITLE
.v file extension clashes with verilog files.

### ISSUE THREAD

**Issue by pierre-rouleau** [vlang/v-emacs#3]
It's unfortunate that V uses .v file extension.  Verilog uses .v and .vh , clashing with V.  Users interested in V might also be working with hardware and use Verilog.

**Comment by Naheel-Azawy** [comment]
Yep.. It's sad but true, you have to pick. For me, I use verilog once in a while. So I prefer to have it work with V by default.

```emacs-lisp
(setq auto-mode-alist
      (cons
       '("\\.v$" . vlang-mode)
       auto-mode-alist))
```

And when needed, `M-x verilog-mode`.

**Comment by pierre-rouleau** [comment]
Thanks for the reply.
Ideally the content of the file would be used to distinguish them, at least when visiting the file.
BTW.  Is there a vlang-mode code somewhere? The repo seems empty.

**Comment by Naheel-Azawy** [comment]
@pierre-rouleau Yeah, I made one few days before this reapo was created and I named it [vlang-mode](https://github.com/Naheel-Azawy/vlang-mode.el). You can check issue #1. However, I didn't see much interest so I didn't put much effort into it, only basic syntax highlighting. About a month later, [v-mode](https://github.com/damon-kwok/v-mode) was created. This appears to have much more features so you might be more interested in it. Perhaps open an issue there? I think it won't be so hard to implement that.

**Comment by pierre-rouleau** [comment]
@Naheel-Azawy Thanks. Why would you rename clang-mode to v-emacs?   Personally I prefer vlang-mode.  Is it because you wanted to implement something different as vlang-mode?  The 'v' letter could mean anything, like 'v' for victory. vlang is more explicit.

**Comment by pierre-rouleau** [comment]
@Naheel-Azawy .  The only thing I would do, though is rename your project from vlang-mode.el to just "vlang-mode".

**Comment by pierre-rouleau** [comment]
@Naheel-Azawy , just submitted a PR to prevent a byte-compiler warning.  I'm going to include this in my project (PEL) which will install vlang-mode automatically if a user-option is set accordingly.  I know this is all experimental but that aspect of PEL is also experimental.  I have not yet really started learning V but when I do I'll look at the mode setting and might want to add more to  it if I have time.

**Comment by Naheel-Azawy** [comment]
> @Naheel-Azawy Thanks. Why would you rename clang-mode to v-emacs? Personally I prefer vlang-mode. Is it because you wanted to implement something different as vlang-mode? The 'v' letter could mean anything, like 'v' for victory. vlang is more explicit.

Yeah I agree. I didn't want to change the name. I just found this repo and I thought it would be easier for people to find it here under vlang.

> @Naheel-Azawy . The only thing I would do, though is rename your project from vlang-mode.el to just "vlang-mode".

Done.

> @Naheel-Azawy , just submitted a PR to prevent a byte-compiler warning. I'm going to include this in my project (PEL) which will install vlang-mode automatically if a user-option is set accordingly. I know this is all experimental but that aspect of PEL is also experimental. I have not yet really started learning V but when I do I'll look at the mode setting and might want to add more to it if I have time.

Thanks for your PR! And that PEL looks cool, will check it out!

**Comment by pierre-rouleau** [comment]
Thanks for updating the name.  I have integrated your package into PEL (along with v-mode which has more features).  If I have time I'll look at both and will  try to learn more about writing modes and writing V.

**Comment by Naheel-Azawy** [comment]
@pierre-rouleau Thanks!
And sorry to @vlang we went a bit off topic.
